src/formatting_stack/strategies/impl.clj

```diff
@@ -17,8 +18,40 @@
 ;; modified, added
 (def git-not-completely-staged-regex #"^( M|AM|MM|AD| D|\?\?|) ")
 
-(defn file-entries [& args]
-  (->> args (apply sh) :out str/split-lines (filter seq)))
+(speced/def-with-doc ::file-entry
+  "Not necessarily a filename,
+  e.g. the string \"M  src/formatting_stack/strategies/impl.clj\" is a valid output."
+  string?)
+
+(spec/def ::file-entries (spec/coll-of ::file-entry))
+
+(speced/defn ^::file-entries file-entries
+  [& args]
+  (->> args (apply sh) :out string/split-lines (filter seq)))
+
+(def separator-pattern (re-pattern File/separator))
+
+(def ^:dynamic *skip-existing-files-check?* false)
+
+(spec/def ::existing-files (spec/coll-of (speced/fn [^string? s]
+                                           (if *skip-existing-files-check?*
+                                             true
+                                             (-> s File. .exists)))))
```

Comment:
optional `(or *skip-existing-files-check?* (-> s File. .exists))`

Reply:
Doable, but as an habit I (we?) tend to favor booleans as predicate return values

If `*skip-existing-files-check?*` was `42`, this fn would return 42, failing at that boolean-ness